### DISCUSSION TITLE
Trigger step if any previous step fails?

### DISCUSSION THREAD

**Discussion by inwardmovement** [community/community#26189] [Actions]
<p>Is there a way to trigger a step if any of the previous ones failed, not only the previous one?</p>
<p>I could add this “if: failure()” step after each other step, but it would be overloaded…</p>
<p>I’d like to have this feature please.</p>

**Comment by chrispat** [comment]
<p>If: failure() should run regardless of which step failed the given job.</p>

**Comment by inwardmovement** [comment]
<p>It is not the case, the scheduled CI of yesterday at midnight in all my repos configured with scheduled CI failed at “Setup Node.js” step and the last step “Notify Discord” of the workflow didn’t fired. I triggered again the run and it passed, but apparently I can’t find the log of the failed version, so I cannot show you…</p>
<p>I’ll write here if it happens again (this time I won’t re-trigger so the log of the failed run will not be overridden), but indeed</p>
<pre><code>if: failure()
</code></pre>
<p><em>generally</em> runs regardless of which step failed. I guess it was an exception/bug.</p>
<p>In case you can investigate, <a href="https://github.com/algosphere/algo-portal/actions?query=workflow%3Aci-cd-master" rel="nofollow noopener">here are the runs</a>.</p>

**Comment by weide-zhou** [comment]
<p>Hi <a class="mention" href="/inwardmovement">@inwardmovement</a> ,</p>
<p>Thanks for your clarification! This should be an temporary issue which caused by backend services/images update, the ci services exit accidently including the latter steps even you monitor the failure. </p>
<p>Thanks.</p>

**Comment by inwardmovement** [comment]
<p>Ok thanks for the information!</p>

**Comment by inwardmovement** [comment]
<p><a class="mention" href="/weide-zhou">@weide-zhou</a> <a class="mention" href="/chrispat">@chrispat</a> so there it is again, it was ok on push yesterday but fails today on scheduled: <a href="https://github.com/inwardmovement/mettaconseil/runs/471229159?check_suite_focus=true" rel="nofollow noopener">log</a> </p>
<p>Any idea? Is it a problem with the actions themselves or with GitHub’s “CI engine”?</p>

**Comment by inwardmovement** [comment]
<p>Nevermind, I will switch to email notifications for CI failures as notifications depending on workflow actions are not reliable enough in a pro context (and this is not depending on GitHub but on actions developers I guess).</p>

**Comment by weide-zhou** [comment, marked as answer]
<p>Hi <a class="mention" href="/inwardmovement">@inwardmovement</a> ,</p>
<p>It’s caused by the action itself.  The ‘netlify/open-api’ has released v0.13.0 just 21hours ago, and it’s incompatible with node 10, that’s why the error occurs. Now a new release v0.13.1 has been created, and removed ‘engines.node’. Please try again for your workflow.</p>
<p>You can refer to the <a href="https://github.com/netlify/open-api/releases" rel="nofollow noopener">release</a> here for more info.</p>
<p>Thanks.</p>
